pgjdbc/src/main/java/org/postgresql/jdbc/PgCallableStatement.java

```diff
@@ -691,7 +696,132 @@ public Reader getCharacterStream(String parameterName) throws SQLException {
   }
 
   public <T> T getObject(int parameterIndex, Class<T> type) throws SQLException {
-    if (type == ResultSet.class) {
+    if (type == null) {
+      throw new SQLException("type is null");
+    }
+    checkIndex(parameterIndex);
+    int sqlType = functionReturnType[parameterIndex - 1];
+    if (type == BigDecimal.class) {
+      if (sqlType == Types.NUMERIC || sqlType == Types.DECIMAL) {
+        return type.cast(getBigDecimal(parameterIndex));
+      } else {
+        throw new PSQLException(GT.tr("Unsupported type conversion to {1}.", type),
+                PSQLState.INVALID_PARAMETER_VALUE);
+      }
+    } else if (type == String.class) {
+      if (sqlType == Types.CHAR || sqlType == Types.VARCHAR) {
+        return type.cast(getString(parameterIndex));
+      } else {
+        throw new PSQLException(GT.tr("Unsupported type conversion to {1}.", type),
+                PSQLState.INVALID_PARAMETER_VALUE);
+      }
+    } else if (type == Boolean.class) {
+      if (sqlType == Types.BOOLEAN || sqlType == Types.BIT) {
+        boolean booleanValue = getBoolean(parameterIndex);
+        if (wasNull()) {
+          return null;
+        }
+        return type.cast(booleanValue);
+      } else {
+        throw new PSQLException(GT.tr("Unsupported type conversion to {1}.", type),
+                PSQLState.INVALID_PARAMETER_VALUE);
+      }
+    } else if (type == Integer.class) {
+      if (sqlType == Types.SMALLINT || sqlType == Types.INTEGER) {
```

Comment:
You are right, this is the function return type, so no TINYINT here, I just noted that is a getObject not a setObject like in https://github.com/pgjdbc/pgjdbc/blob/cd77693ca22924479a39b8d925f276879023672a/pgjdbc/src/main/java/org/postgresql/jdbc/PgPreparedStatement.java#L549

Well, what happen if I ask for a Short.class? should not be covered?